### PR TITLE
[CSR-2066] feat: added full test suite generation for convert command

### PR DESCRIPTION
**Pull request by miguelangaranocurrents**
## Description

> Provide a brief description of the changes in this PR. Important to list all the changes made in overall. Describe any improvements, follow up tasks or edge cases related to this PR

This PR add the fullTestSuite.json file generation to the convert command.
It also allows generating multiple groups for the full test suite by accepting multiple files as different test groups.

## PR Checklist

- [x] I performed manual tests or added tests that prove my fix is effective or that my feature works.
- [x] I have performed a self-review of my code.
- [x] I have annotated my PR with comments, particularly in hard-to-understand areas.
- [x] I have considered the security implications of this work.

## Release Plan

> Do we need to update any environment variables? Is there any order of releases required?

- Package release

## Demo

> Add screenshots or videos demonstrating the solution if applicable

https://www.loom.com/share/02ef05e48df64832a7cb884c960db874?sid=60ea25b0-e032-4e73-b888-1d0b0dffe6c3

## Manual Testing

> Describe in steps (support it with images if needed) how to try the changes of this PR. (Even the start/setup of the services needed to try it out). If it's a bug also include reproduction steps

1. Use the convert command to create instance files and the fullTestSuite JSON file
2. Use multiple JUnit XML files for having multiple groups
3. Use the upload command

### CODE REVIEW

File: packages/cmd/src/services/upload/index.ts [hunk 1]
```diff
@@ -223,7 +223,7 @@ function getMarkerFilePath(reportDir: string) {
   return path.join(reportDir, 'upload.marker.json');
 }
 
-function getFullTestSuiteFilePath(reportDir: string) {
+export function getFullTestSuiteFilePath(reportDir: string) {
```
[the remark this answers is not in the document]
Comment (vCaisim) [reply packages/cmd/src/services/upload/index.ts]: In `packages/cmd/src/services/upload/discovery/junit/scanner.ts` change the import of `getFullTestSuiteFilePath` to `import { getFullTestSuiteFilePath } from '../../path';`

File: packages/cmd/src/lib/fs.ts [hunk 2]
```diff
@@ -24,6 +24,27 @@ export async function writeFileAsync(filePath: string, content: string) {
...
+) {
+  try {
+    // Check if file exists
+    try {
+      await fs.access(filePath);
+      // If we reach here, file exists
+      return filePath;
+    } catch {
+      // File doesn't exist, create it
+      await writeFileAsync(filePath, content);
```
Comment (vCaisim) [inline packages/cmd/src/lib/fs.ts:39]: better use ` await fs.writeFile(filePath, content, 'utf8');` because if `writeFileAsync` fails, 2 identical error messages will be displayed

File: packages/cmd/src/services/convert/getFullTestSuite.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+} from '../upload/discovery';
+import { TestCase, TestSuite, TestSuites } from './types';
+import {
+  ensureArray,
+  generateTestId,
+  getSuiteName,
+  getTestTitle,
+} from './utils';
+
+export async function getFullTestSuite(parsedXMLInputs: TestSuites[]) {
```
Comment (vCaisim) [inline packages/cmd/src/services/convert/getFullTestSuite.ts:14]: no need for `async`

Just a suggestion: since it creates the `FullTestSuite` from `TestSuites[]`, maybe use another name, like `createFullTestSuite` or something similar

File: packages/cmd/src/services/convert/getParsedXMLArray.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  );
+
+  const parsedXMLInputs = (
+    await Promise.all(filesData.map(getParsedXMLInput))
+  ).filter(Boolean);
+
+  return parsedXMLInputs;
+}
+
+const getParsedXMLInput = async (XMLString: string) => {
```
Comment (vCaisim) [inline packages/cmd/src/services/convert/getParsedXMLArray.ts:19]: - `getParsedXMLInput`, don't trim 2 times, use a variable
- it returns `undefined` if `XMLString` is emtpy string or `parsedXMLInput` is falsy value and `getParsedXMLArray` ends up with `[undefined]`

**Review by vCaisim — CHANGES_REQUESTED** [review]
Tested, the issue is solved. Please address the latest feedback and update the option description for the `--input-file` so it is clear that it can accept multiple paths
